### PR TITLE
Dockerfile: Install sparse package for code analysis

### PR DESCRIPTION
**Pull request by aiChaoSONG**
The sparse tool can be used to do static code analysis,
which helps to find coding faults at compile-time.

Signed-off-by: Chao Song <chao.song@linux.intel.com>

The Sound Firmware Team will need this package for sparse test. I think zephyr should do sparse test now or sooner.

We have to clone the latest code and compile, the sparse package shipped with ubuntu 20.04 can not work for us.

@lyakh @keqiaozhang FYI

### CODE REVIEW

**Comment by aiChaoSONG** [comment]
I see this error, `Error: The log was not found. It may have been deleted based on retention settings.` can someone tell me what happened, how can i fix this?

**Comment by cjgraydon** [comment]
Looks like the CI workflow for the PR failed due to some kind of runner/network issue: https://github.com/zephyrproject-rtos/docker-image/actions/runs/2588361237

i.e. [Build](https://github.com/zephyrproject-rtos/docker-image/runs/7133379554?check_suite_focus=true)
The hosted runner: GitHub Actions 15 lost communication with the server. Anything in your workflow that terminates the runner process, starves it for CPU/Memory, or blocks its network access can cause this error.

It looks like the run timed out and it never made it far enough to generate the log that subsequent workflow steps want. The step before where you see the log error message threw some kind of error.

You might try updating your PR or doing an empty commit to try to trigger another run: git commit --allow-empty -m "this is an empty commit"    Pushing that should force CI to run again, in theory. Might just be something flukey that caused the initial run to time out.

**Comment by aiChaoSONG** [comment]
@cjgraydon Thank you, but looks I am first-time contributor, need approval to run GH action.

**Comment by aiChaoSONG** [comment]
failed at Installing uefi-run utility, not even touch my changes.

**Comment by stephanosio** [comment]
It looks like the resources available on the default GitHub-provided CI runner is no longer sufficient to build the docker image. I will see if I can rework the CI to fix this.

**Comment by cjgraydon** [comment]
> @cjgraydon Thank you, but looks I am first-time contributor, need approval to run GH action.

Side note: Any additional push to your PR should trigger a new CI run. So you could dodge having to have direct access to the GitHub actions. I'd think an empty commit to your PR would do the trick, but I might be off on that.  I use this approach in our enterprise GH on occasion. Might be different in public GH, though?  Anyway, just a little more detail on what I was talking about in case I wasn't especially clear. :)

**Review by marc-hb — CHANGES_REQUESTED** [review]
A record of which particular sparse SHA1 was used must be kept somewhere. I don't know how that's normally done, maybe you can add a new `ARG SPARSE_COMMIT` variable at the top and check that out?

**Comment by aiChaoSONG** [comment]
@stephanosio Is CI good now? please help to approve and test this patch, thank you.

**Comment by aiChaoSONG** [comment]
@stephanosio Thank you, the test passed, will you push a new image to docker hub after this merged?

File: Dockerfile
Comment (stephanosio) [inline Dockerfile:10]: to make it consistent with the rest

```suggestion
ARG LLVM_VERSION=12
ARG BSIM_VERSION=v1.0.3
ARG SPARSE_VERSION=v0.6.4
ARG WGET_ARGS="-q --show-progress --progress=bar:force:noscroll --no-check-certificate"
```

File: Dockerfile
Comment (stephanosio) [inline Dockerfile:178]:
```suggestion
	cd sparse && git checkout ${SPARSE_VERSION} && \
```

**Review by stephanosio — CHANGES_REQUESTED** [review]
unless you actually need a feature from the unstable/development branch, we should use a release tag.

File: Dockerfile
Comment (aiChaoSONG) [reply Dockerfile:178]: @stephanosio Yes, we need the feature from the development branch. I understand that there are sparse tar.gz releases, but the latest sparse  0.6.4 doesn't contain the feature we need. that's why I have to use the development branch and specify a commit ID.
But I can promise you that once there is a new release or a new tag that covers the feature we need, I will submit a new PR.

**Review by stephanosio — APPROVED** [review]

File: Dockerfile
Comment (aiChaoSONG) [reply Dockerfile]: updated, thank you

**Pull request merged**
